docs/Coding-Guidelines/SelfContained.md

```diff
@@ -0,0 +1,46 @@
+# Self-Contained
+
+## What is 'Self-Contained'?
+
+Windows App SDK can be deployed via MSIX or self-contained, aka WinAppSDK/MSIX and WinAppSDK/SelfContained.
+
+See the [Self-Contained spec](https://github.com/microsoft/WindowsAppSDK/pull/1202) for more details.
+
+## How to detect if Windows App SDK is deployed Self-Contained?
+
+If necessary, use `Microsoft::WindowsAppSDK::SelfContained::IsSelfContained()` to detect if the
+current process is using WindowsAppSDK deployed via SelfContained (vs MSIX).
```

Comment:
But you can have an app that's MSIX packaged and self-contained, right? Perhaps drop the parenthetical to avoid having to explain.

```suggestion
current process is using WindowsAppSDK deployed via SelfContained.
```

Reply:
> But you can have an app that's MSIX packaged and self-contained, right?

Yes.

There are 2 actors involved -- (1) WinAppSDK bits and (2) an app using WinAppSDK. This IsSelfContained() determines how (1) WinAppSDK is deployed, whether that's via MSIX packages or as files with/in/part-of your app.

That's independent of how (2) an app is deployed. You absolutely can make MyApp.msix using WinAppSDK/SelfContained, in which case MyApp.msix contains MyApp.exe etc AND microsoft.windowsappruntime.dll and other files making up WinAppSDK's runtime.

@mikebattista Terminology seems to be a bit confusing. I look forward to [the Self-Contained spec](https://github.com/microsoft/WindowsAppSDK/pull/1202) defining the terminology we should use.